ui/app/routes/vault/cluster/secrets/backend/configuration/index.js

```diff
@@ -73,112 +37,121 @@ export default class SecretsBackendConfigurationRoute extends Route {
         return this.fetchGcpConfig(id);
       case 'ssh':
         return this.fetchSshCaConfig(id);
-      default:
-        return reject({ httpStatus: 404, message: 'not found', path: id });
     }
   }
 
-  async fetchAwsConfigs(id) {
+  async fetchAwsConfigs(path) {
     // AWS has two configuration endpoints root and lease, as well as a separate endpoint for the issuer.
-    // return an array of these responses.
-    const configArray = [];
-    const configRoot = await this.fetchAwsConfig(id, 'aws/root-config');
-    const configLease = await this.fetchAwsConfig(id, 'aws/lease-config');
-    let issuer = null;
-    if (this.version.isEnterprise && configRoot) {
-      // issuer is an enterprise only related feature
-      // issuer is also a global endpoint that doesn't mean anything in the AWS secret details context if WIF related fields on the rootConfig have not been set.
-      const WIF_FIELDS = ['roleArn', 'identityTokenAudience', 'identityTokenTtl'];
-      WIF_FIELDS.some((field) => configRoot[field]) ? (issuer = await this.fetchIssuer()) : null;
-    }
-    configArray.push(configRoot, configLease, issuer);
-    return configArray;
-  }
-
-  async fetchAwsConfig(id, modelPath) {
-    try {
-      return await this.store.queryRecord(modelPath, { backend: id });
-    } catch (e) {
-      if (e.httpStatus === 404) {
+    const handleError = async (e) => {
+      const error = await this.parseApiError(e);
+      if (error.httpStatus === 404) {
         // a 404 error is thrown when the lease config hasn't been set yet.
-        return;
+        return {};
       }
-      throw e;
+      throw error;
+    };
+
+    const { data: configRoot } = await this.api.secrets
+      .awsReadRootIamCredentialsConfiguration(path)
+      .catch(handleError);
+    const { data: configLease } = await this.api.secrets.awsReadLeaseConfiguration(path).catch(handleError);
+
+    const WIF_FIELDS = ['roleArn', 'identityTokenAudience', 'identityTokenTtl'];
+    const issuer = await this.checkIssuer(configRoot, WIF_FIELDS);
+
+    if (configRoot) {
+      return Object.assign(configRoot, configLease, issuer);
```

Comment:
Are we sure these objects won't have overlapping keys that would be accidentally overridden by this assignment? Just want to double check since these were previously an array of objects. Returning an object makes much more sense, though!

Reply:
That's a good question! I double checked the original models and they have different properties.